sanic/response.py

```diff
@@ -142,13 +141,62 @@ def cookies(self):
         return self._cookies
 
 
+class ContentRangeHandler:
+    """
+    This class is for parsing the request header
+    """
+    __slots__ = ('start', 'end', 'size', 'total', 'headers')
+
+    def __init__(self, request, stats):
+        self.start = self.size = 0
+        self.end = None
+        self.headers = dict()
+        self.total = stats.st_size
+        _range = request.headers.get('Range')
+        if _range:
+            self.start, self.end = ContentRangeHandler.parse_range(_range)
+            if self.start is not None and self.end is not None:
+                self.size = self.end - self.start
+            elif self.end is not None:
+                self.size = self.end
+            elif self.start is not None:
+                self.size = self.total - self.start
+            else:
+                self.size = self.total
+            self.headers['Content-Range'] = "bytes %s-%s/%s" % (
+                self.start, self.end, self.total)
+        else:
+            self.size = self.total
+
+    def __bool__(self):
+        return self.size > 0
+
+    @staticmethod
+    def parse_range(range_header):
+        unit, _, value = tuple(map(str.strip, range_header.partition('=')))
+        if unit != 'bytes':
+            return None
+        start_b, _, end_b = tuple(map(str.strip, value.partition('-')))
+        try:
+            start = int(start_b) if start_b.strip() else None
+            end = int(end_b) if end_b.strip() else None
+        except ValueError:
+            return None
```

Comment:
Returning None here when you expect to return a tuple could prove problematic and would raise a `TypeError`